source/components/ProgressBarComponent.mc

```diff
@@ -0,0 +1,40 @@
+import Toybox.Application;
+import Toybox.Graphics;
+import Toybox.Lang;
+import Toybox.Time;
+import Toybox.WatchUi;
+
+class ProgressBarComponent extends WatchUi.Drawable {
+  hidden var mController as Types.Controllers.EverythingController;
+  hidden var mUpdateInterval as Time.Duration;
+  hidden var mUpdated as Time.Moment;
+  hidden var mProgress as Numeric;
+
+  function initialize(params as Types.Components.ProgressBarParams) {
+    Drawable.initialize(params);
+    mController = Utils.Controller.getController(params[:controller] as Types.Controllers.Id);
+    mUpdated = new Time.Moment(0);
+    mUpdateInterval = Utils.Component.getUpdateInterval(params[:updateInterval]);
```

Comment:
If we added a `BaseComponent`, we could remove these reoccurring lines from every component.